Adding basic documentation to go along with the openid-connect example

filmaj opened the pull request:
Fixes #1356

Very basic first pass at this... is this sufficient? What is it missing?

srajiang commented on docs/_packages/web_api.md:
```diff
@@ -682,3 +682,16 @@ const clientSecret = process.env.SLACK_CLIENT_SECRET;
+
+### Sign in with Slack via OpenID Connect
+
+By leveraging two APIs provided by `@slack/web-api`, [`openid.connect.token`](https://api.slack.com/methods/openid.connect.token)
```
```suggestion
**Sign in With Slack** via OpenID Connect gives users the ability to sign into your service using their Slack profile.

The `@slack/web-api` package supports the following API methods which you can use to implement **Sign in With Slack**: 

- [`openid.connect.token`](https://api.slack.com/methods/openid.connect.token)

```

srajiang commented on docs/_packages/web_api.md:
```diff
@@ -682,3 +682,16 @@ const clientSecret = process.env.SLACK_CLIENT_SECRET;
+### Sign in with Slack via OpenID Connect
+
+By leveraging two APIs provided by `@slack/web-api`, [`openid.connect.token`](https://api.slack.com/methods/openid.connect.token)
+and [`openid.connect.userInfo`](https://api.slack.com/methods/openid.connect.userInfo), you can provide users the
```
```suggestion

- [`openid.connect.userInfo`](https://api.slack.com/methods/openid.connect.userInfo).

```

srajiang commented on docs/_packages/web_api.md:
```diff
@@ -682,3 +682,16 @@ const clientSecret = process.env.SLACK_CLIENT_SECRET;
+and [`openid.connect.userInfo`](https://api.slack.com/methods/openid.connect.userInfo), you can provide users the
+ability to sign into your service using their Slack profile.
+
+A fully functioning [application example is available](https://github.com/slackapi/node-slack-sdk/blob/main/examples/openid-connect) for your perusal!
```
```suggestion
Here's a fully-functioning [sample application implementation](https://github.com/slackapi/node-slack-sdk/blob/main/examples/openid-connect) for your perusal!
```

srajiang commented on docs/_packages/web_api.md:
```diff
@@ -682,3 +682,16 @@ const clientSecret = process.env.SLACK_CLIENT_SECRET;
+
+A fully functioning [application example is available](https://github.com/slackapi/node-slack-sdk/blob/main/examples/openid-connect) for your perusal!
+
+Full details on how the Sign in with Slack flow works, application setup, as well as helpful resources like a Sign in With Slack
```
```suggestion
💡 To read more about how **Sign in with Slack** works, and to access helpful resources like a Sign in With Slack
```

srajiang commented on docs/_packages/web_api.md:
```diff
@@ -682,3 +682,16 @@ const clientSecret = process.env.SLACK_CLIENT_SECRET;
+A fully functioning [application example is available](https://github.com/slackapi/node-slack-sdk/blob/main/examples/openid-connect) for your perusal!
+
+Full details on how the Sign in with Slack flow works, application setup, as well as helpful resources like a Sign in With Slack
+button generator and other design assets are available on the [Authentication: Sign in with Slack](https://api.slack.com/authentication/sign-in-with-slack)
```
```suggestion
button generator and other design assets, check out: [Authentication: Sign in with Slack](https://api.slack.com/authentication/sign-in-with-slack)
```

srajiang reviewed (changes requested):
Hows this for a slightly modified wording?

seratch commented:
How about having a link from the oauth document page as well? https://github.com/slackapi/node-slack-sdk/issues/1356#issuecomment-961461320

filmaj commented:
@srajiang thank you for your suggestions, I have incorporated them!

@seratch I added a 'limitations' section to the oauth package docs calling out the inability to use the package to implement Sign in with Slack, and linked to the new web-api package doc on the topic.

What do you all think of the latest version of these docs?

seratch reviewed (approved):
LGTM
